main/src/main/scala/sbt/internal/Load.scala

```diff
@@ -930,10 +930,7 @@ private[sbt] object Load {
           )
         case Nil if makeOrDiscoverRoot =>
           log.debug(s"[Loading] Scanning directory ${buildBase}")
-          val auto =
-            if (context.globalPluginProject)
-              AddSettings.seq(AddSettings.defaultSbtFiles, AddSettings.sbtFiles(extraSbtFiles: _*))
-            else AddSettings.defaultSbtFiles
+          val auto = AddSettings.seq(AddSettings.defaultSbtFiles, AddSettings.sbtFiles(extraSbtFiles: _*))
```

Comment:
loading mechanism gets called multiple times at different level, so the above if expression is meant to load plugin files only for the metabuild.

Reply:
How many levels are there? What is the intended level for loading plugin files then if it is not the metabuild level, which I presume corresponds to `context.globalPluginProject == true`?

Reply:
sbt build is recursive, and the plugin loading is just build loading (build.sbt) inside `project/` directory. The directory structure can go on. In addition to the working directory, there's also machine-wide global settings, and global plugins. Each subproject can also have its own build.sbt.

Reply:
yes, i understand this principle of sbt; however, I'm still puzzled why the `--addPluginSbtFile=...` option has become ineffective. I thought, perhaps incorrectly, that I had found the reason but perhaps there's something more to it.

Bottom line is simple: when we do: `sbt -sbt-version=1.2.3 --addPluginSbtFile=foo.sbt`, when and where should this option be processed? It's obviously not happening in my environment and I'm trying to figure out why. I've tried lots of things including removing `~/.sbt` entirely and `~/.m2/repository` without luck. The only workaround I found is to move `foo.sbt` to `./project/foo.sbt`.